File: frontend/src/types/auth-types.ts
export type SignupFormType = {
    name: string;
    email: string;
    password: string;
    confirm_password: string;
};

export type SignupFormErrorsType =
    | {
        name?: { errors: string[] };
        email?: { errors: string[] };
        password?: { errors: string[] };
        confirm_password?: { errors: string[] };
    }
    | undefined;

File: server/src/services/auth-service/internal/types/auth-types.go
package types

type UserRegistrationType struct {
	Name            string `json:"name" validate:"required,min=2,max=100"`
	Email           string `json:"email" validate:"required,email" db:"email"`
	Password        string `json:"password" validate:"required,min=8" db:"password"`
	ConfirmPassword string `json:"confirm_password" validate:"required,min=8,eqfield=Password"`
}

type Tokens struct {
	AccessToken  string `json:"token"`
	RefreshToken string `json:"refresh_token"`
}

type GoogleOauthUser struct {
	ID            string `json:"id"`
	Email         string `json:"email"`
	VerifiedEmail bool   `json:"verified_email"`
	Picture       string `json:"picture"`
	Name          string `json:"name"`
	GivenName     string `json:"given_name"`
	FamilyName    string `json:"family_name"`
}
